File: app/src/main/java/pw/janyo/whatanime/activity/HistoryActivity.java
package pw.janyo.whatanime.activity;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.support.v7.widget.helper.ItemTouchHelper;
import android.view.View;

import org.litepal.crud.DataSupport;

import java.util.List;

import pw.janyo.whatanime.R;
import pw.janyo.whatanime.adapter.HistoryAdapter;
import pw.janyo.whatanime.classes.History;
import pw.janyo.whatanime.util.WAFileUtil;

public class HistoryActivity extends AppCompatActivity {
	private Toolbar toolbar;

	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		initialization();
		monitor();
	}

	private void initialization() {
		setContentView(R.layout.activity_history);
		toolbar = findViewById(R.id.toolbar);

		final List<History> list = WAFileUtil.checkList(DataSupport.findAll(History.class));
		RecyclerView recyclerView = findViewById(R.id.recyclerView);
		recyclerView.setLayoutManager(new LinearLayoutManager(HistoryActivity.this));
		final HistoryAdapter adapter = new HistoryAdapter(HistoryActivity.this, list);
		ItemTouchHelper.SimpleCallback callback = new ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.RIGHT | ItemTouchHelper.LEFT) {
			@Override
			public boolean onMove(RecyclerView recyclerView, RecyclerView.ViewHolder viewHolder, RecyclerView.ViewHolder target) {
				return false;
			}

			@Override
			public void onSwiped(RecyclerView.ViewHolder viewHolder, int direction) {
				WAFileUtil.deleteHistory(list.get(viewHolder.getAdapterPosition()));
				list.remove(viewHolder.getAdapterPosition());
				adapter.notifyItemRemoved(viewHolder.getAdapterPosition());
			}
		};
		new ItemTouchHelper(callback).attachToRecyclerView(recyclerView);
		recyclerView.setAdapter(adapter);

		setSupportActionBar(toolbar);
	}

	private void monitor() {
		toolbar.setNavigationOnClickListener(new View.OnClickListener() {
			@Override
			public void onClick(View view) {
				finish();
			}
		});
	}
}

File: app/src/main/java/pw/janyo/whatanime/activity/HistoryActivity.kt
//package pw.janyo.whatanime.activity
//
//import android.os.Bundle
//import androidx.appcompat.app.AppCompatActivity
//import androidx.recyclerview.widget.LinearLayoutManager
//import androidx.recyclerview.widget.RecyclerView
//import androidx.appcompat.widget.Toolbar
//import androidx.recyclerview.widget.ItemTouchHelper
//import android.view.View
//
//import org.litepal.crud.DataSupport
//
//import pw.janyo.whatanime.R
//import pw.janyo.whatanime.adapter.HistoryAdapter
//import pw.janyo.whatanime.classes.History
//import pw.janyo.whatanime.util.WAFileUtil
//
//class HistoryActivity : AppCompatActivity() {
//	private var toolbar: Toolbar? = null
//
//	override fun onCreate(savedInstanceState: Bundle?) {
//		super.onCreate(savedInstanceState)
//		initialization()
//		monitor()
//	}
//
//	private fun initialization() {
//		setContentView(R.layout.activity_history)
//		toolbar = findViewById(R.id.toolbar)
//
//		val list = WAFileUtil.checkList(DataSupport.findAll(History::class.java))
//		val recyclerView = findViewById<RecyclerView>(R.id.recyclerView)
//		recyclerView.layoutManager = LinearLayoutManager(this@HistoryActivity)
//		val adapter = HistoryAdapter(this@HistoryActivity, list)
//		val callback = object : ItemTouchHelper.SimpleCallback(0, ItemTouchHelper.RIGHT or ItemTouchHelper.LEFT) {
//			override fun onMove(recyclerView: RecyclerView, viewHolder: RecyclerView.ViewHolder, target: RecyclerView.ViewHolder): Boolean {
//				return false
//			}
//
//			override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
//				WAFileUtil.deleteHistory(list[viewHolder.adapterPosition])
//				list.removeAt(viewHolder.adapterPosition)
//				adapter.notifyItemRemoved(viewHolder.adapterPosition)
//			}
//		}
//		ItemTouchHelper(callback).attachToRecyclerView(recyclerView)
//		recyclerView.adapter = adapter
//
//		setSupportActionBar(toolbar)
//	}
//
//	private fun monitor() {
//		toolbar!!.setNavigationOnClickListener { finish() }
//	}
//}
